vm/http_client.go

```diff
@@ -0,0 +1,119 @@
+package vm
+
+import (
+	"strconv"
+	"net/http"
+	"errors"
+	"strings"
+	"fmt"
+	"io/ioutil"
+	gerrors "github.com/goby-lang/goby/vm/errors"
+)
+
+func builtinHTTPClientClassMethods() []*BuiltInMethodObject {
+	goClient := http.DefaultClient
+
+	return []*BuiltInMethodObject{
+		{
+			// Sends a GET request to the target and returns the HTTP response as a string.
+			Name: "send",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+
+					if len(args) != 0 {
+						return t.vm.initErrorObject(gerrors.ArgumentError, "Expect 0 arguments. got=%v", strconv.Itoa(len(args)))
+
+					}
+
+					req := httpRequestClass.initializeInstance()
+
+					result := t.builtInMethodYield(blockFrame, req)
+
+					if err, ok := result.Target.(*Error); ok {
+						fmt.Printf("Error: %s", err.Message)
+						return err //a Error object
+					}
+
+
+					goReq, err := requestGobyToGo(req)
+					if err != nil {
+						return t.vm.initErrorObject(gerrors.ArgumentError, "Request object incomplete object %s", err)
+					}
+
+					resp, err := goClient.Do(goReq)
+					if err != nil {
+						fmt.Println("do error: ", err)
+						return t.vm.initErrorObject(gerrors.InternalError, "Could not get response: %s", err)
+					}
+
+					gobyResp, err := responseGoToGoby(t, resp)
+					if err != nil {
+						return t.vm.initErrorObject(gerrors.InternalError, "Could not read response: %s", err)
+					}
+
+					return gobyResp
+				}
+			},
+		},
+	}
+}
+
+func requestGobyToGo(gobyReq *RObject) (*http.Request, error) {
+	//:method, :protocol, :body, :content_length, :transfer_encoding, :host, :path, :url, :params
+	uObj, ok := gobyReq.instanceVariableGet("@url")
+	if !ok {
+		return nil, errors.New("could not get url")
+	}
+
+	u := uObj.(*StringObject).value
+
+	methodObj, ok := gobyReq.instanceVariableGet("@method")
+	if !ok {
+		return nil, errors.New("could not get method")
```

Comment:
You can just use `fmt.Errorf` to return an error object, so you can avoid using the builtin errors package.